src/views/ApartmentApp/components/StaffMenu/components/ApplicationTable/components/ApplicationRow/index.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Collapse, IconButton, TableCell, TableRow } from '@material-ui/core/';
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
+import ApplicantSubTable from './components/ApplicantSubTable';
+import HallSubTable from './components/HallSubTable';
+
+const ApplicationRow = ({ applicationDetails, labelId }) => {
+  const [open, setOpen] = React.useState(false);
+
+  return (
+    <React.Fragment>
+      <TableRow key={applicationDetails.ApplicationID} onClick={() => setOpen(!open)}>
+        <TableCell align="center" component="th" id={labelId} scope="row">
+          {applicationDetails.ApplicationID}
+        </TableCell>
+        <TableCell align="center">{applicationDetails.EditorUsername}</TableCell>
+        <TableCell align="center">{applicationDetails.Applicants.length}</TableCell>
```

Comment:
I think we're fairly confident this should always be an array, but it might be useful to add some null safety to this:
```suggestion
        <TableCell align="center">{applicationDetails.Applicants?.length ?? 0}</TableCell>
```

Reply:
That is a very good check to add, because I did have weird problems with it being null or undefined, even momentarily, causing needless error messages
Feel free to commit this as well